lib/blob.js

```diff
@@ -1044,17 +1193,957 @@ Blob.prototype.leaseContainer = function leaseContainer(name, options) {
     headers['x-ms-proposed-lease-id'] = options.proposedLeaseId;
   }
 
+  utils.setConditionalHeaders(headers, {
+    ifModifiedSince: options.ifModifiedSince,
+    ifUnmodifiedSince: options.ifUnmodifiedSince
+  });
+
   return this.request('PUT', path, query, headers).then(function(response) {
     if (response.statusCode !== 200 && response.statusCode !== 201 && response.statusCode !== 202) {
       throw new Error("leaseContainer: Unexpected statusCode: " + response.statusCode);
     }
 
     var result = {
-      leaseId: response.headers['x-ms-lease-id']
+      leaseId: response.headers['x-ms-lease-id'],
+      eTag: response.headers['etag'],
+      lastModified: response.headers['last-modified']
     };
     if (response.headers['x-ms-lease-time']) {
       result.leaseTime = response.headers['x-ms-lease-time'];
     }
     return result;
   });
+};
+
+/**
+ * Creates a new block, page, or append blob, or updates the content of an existing block blob.
```

Comment:
So after reading it seems `putBlob` has very few blobtype specific parameters.

I suspect the only real win with using different method is that it would be easier to document what people should do next. Ie. after calling `putBlockBlob` don't you have to upload blocks and commit them, or maybe that's not required...

Hmm... perhaps nvm this, having one `putBlob` is probably fine..

In a future PR we can add a utility function that does streaming uploads for block blobs with parallelism, retry and commit at the end... Ie. a utility function for doing high-level stuff.